extension/public/reddit.js

```diff
@@ -123,21 +134,40 @@ var Reddit = {
   // retrive the posts iamge link
   getDOMImageLink: function (_post) {
     classSelector = "._2_tDEnGMLxpM6uOa2kaDB3";
-    var url = _post.querySelector(classSelector).getAttribute("src");
-    console.log("image url: " + url); // DEBUG
+    var url = _post.querySelector(classSelector);
+    if(url != null){
+      url = url.getAttribute("src");
+      console.log("image url: " + url); // DEBUG
+      return url;
+    }else{
+      return "n/a";
```

Comment:
This is a non-standard way to throw an error

Reply:
I'd recommend taking a look at this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw

Reply:
"n/a" was not meant to be an error handler, so I've changed the logic on this